nilearn/regions/signal_extraction.py

```diff
@@ -15,8 +15,144 @@
 from .. import masking
 from ..image import new_img_like
 
+INF = 1e-9
+
+
+def _check_shappe_affine_label_img(labels_img, target_shape, target_affine):
+    """Validate shapes and affines of labels.
+    Parameters
+    ----------
+    labels_img : Niimg-like object
+        See http://nilearn.github.io/manipulating_images/input_output.html
+        regions definition as labels. Encodes the region labels of the signals.
+
+    target_shape : numpy.ndarray
+        Desired shape of label images.
+
+    target_affine : numpy.ndarray
+        Desired affine of label images.
+
+    """
+    if labels_img.shape != target_shape:
+        raise ValueError("label_imgs.shape and target_shape must be identical.")
+    if (labels_img.affine.shape != target_affine.shape
+        or abs(labels_img.affine - target_affine).max() > INF):
+        raise ValueError("label img and imgs affines must be identical.")
+
+
+def _check_shappe_affine_maps_masks(target_shape, target_affine,
+                                    img=None, dim=None):
+    """Validate shapes and affines of maps and masks.
+    Parameters
+    ----------
+    target_shape : tuple
+        Desired shape of maps or masks.
+
+    target_affine : numpy.ndarray
+        Desired affine of maps or masks.
+
+    img : Niimg-like object, optional
+        Contains map or mask.
+        See http://nilearn.github.io/manipulating_images/input_output.html
+
+    dim : integer, optional
+        Integer slices a mask for a specific dimension. Default=None.
+
+    Returns
+    -------
+    non_empty : bool,
+        Is only true for non-empty img.
+        
+    """
+    state = False
+
+    # Check shape
+    if img is not None and dim is None:
+        img = _utils.check_niimg_3d(img)
+        if img.shape != target_shape:
+            raise ValueError("mask/map and imgs shapes must be identical.")
+    elif img is not None and img.shape[:dim] != target_shape:
+        raise ValueError("mask/map and imgs shapes must be identical.")
+
+    # Check affines & set state
+    if img is not None and (
+       img.affine.shape != target_affine.shape or
+       abs(img.affine - target_affine).max() > INF):
+        raise ValueError("mask/map and imgs affines must be identical.")
```

Comment:
PEP8 is complaining about the binary operators at the end of the lines in this if condition.

Reply:
Putting the operators at the beginning should be fine. We have set PEP8 to ignore those.